install/gcp/deployment_manager/istio-cluster.jinja

```diff
@@ -129,12 +138,68 @@ resources:
           helm template install/kubernetes/helm/istio --name istio --namespace istio-system $ISTIO_OPTIONS  > istio.yaml
 
           kubectl apply -f istio.yaml
-          kubectl label namespace default istio-injection=enabled
 
-          sleep 150
+          {% if  properties['enableAutomaticSidecarInjection'] %}
+            kubectl label namespace default istio-injection=enabled
+          {% endif %}
+
+          kubectl rollout status deployment.extensions/istio-pilot -n istio-system --timeout=60s
+          kubectl rollout status deployment.extensions/istio-citadel -n istio-system --timeout=60s
+          kubectl rollout status deployment.extensions/istio-policy -n istio-system --timeout=60s
+          kubectl rollout status deployment.extensions/istio-galley -n istio-system --timeout=60s
+          kubectl rollout status deployment.extensions/istio-ingressgateway -n istio-system --timeout=60s
+
+          {% if  properties['enableKiali'] %}
+
+            export USERNAME=$(echo -n '{{  properties['kialiUsername'] }}' | base64)
+            export PASSPHRASE=$(echo -n '{{  properties['kialiPassword'] }}' | base64)
+            export NAMESPACE=istio-system
+
+            echo '
+            apiVersion: v1
+            kind: Secret
+            metadata:
+              name: kiali
+              namespace: $NAMESPACE
+              labels:
+                app: kiali
+            type: Opaque
+            data:
+              username: $USERNAME
+              passphrase: $PASSPHRASE
+            ' | envsubst > kaili_secret.yaml
+
+            kubectl apply -f kaili_secret.yaml
 
-          kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml
-          kubectl apply -f samples/bookinfo/networking/bookinfo-gateway.yaml
+            export KIALI_OPTIONS=" --set kiali.enabled=true "
+
+            {% if  properties['enableGrafana'] %}
+              KIALI_OPTIONS=$KIALI_OPTIONS"  --set kiali.dashboard.grafanaURL=http://$(kubectl get svc grafana -n istio-system -o jsonpath='{.spec.clusterIP}'):3000"
+            {% endif %}
+
+            {% if  properties['enableTracing'] %}
+              KIALI_OPTIONS=$KIALI_OPTIONS" --set kiali.dashboard.jaegerURL=http://$(kubectl get svc tracing -n istio-system -o jsonpath='{.spec.clusterIP}'):80"
+            {% endif %}
+
+            helm template install/kubernetes/helm/istio --name istio --namespace istio-system $KIALI_OPTIONS  > istio_kiali.yaml
+
+            kubectl apply -f istio_kiali.yaml
+
+          {% endif %}
+
+          {% if  properties['enableBookInfoSample'] %}
+
+            kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml
+            kubectl apply -f samples/bookinfo/networking/bookinfo-gateway.yaml
+
+            kubectl rollout status deployment.extensions/productpage-v1 --timeout=60s
```

Comment:
what happens if the timeout occurs?  I don't see any sort of failure detection.  I am not familiar with GCP DM and if it runs these scripts with -x (or what it would do with a prematurely terminating deployment).

60 seconds seems awfully tight.

Reply:
Addin in some bash if-then statement to check for return code.  If any of the deployments fails or times out, the DM itself will fail.  Unfortunately, there is no easy  way to surface any error message back to the user

re: `60s`, ok i increased to `120s` as it took a bit more than 1 min
```
Dec 26 17:58:52 istio-my-cluster-vm startup-script: INFO startup-script: destinationrule.networking.istio.io/istio-telemetry created
Dec 26 17:58:52 istio-my-cluster-vm startup-script: INFO startup-script: + kubectl label namespace default istio-injection=enabled
Dec 26 17:58:52 istio-my-cluster-vm startup-script: INFO startup-script: namespace/default labeled
Dec 26 17:58:52 istio-my-cluster-vm startup-script: INFO startup-script: + kubectl rollout status deployment.extensions/istio-pilot -n istio-system --timeout=120s
Dec 26 17:58:53 istio-my-cluster-vm startup-script: INFO startup-script: Waiting for deployment "istio-pilot" rollout to finish: 0 of 1 updated replicas are available...
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: deployment "istio-pilot" successfully rolled out
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: + kubectl rollout status deployment.extensions/istio-citadel -n istio-system --timeout=120s
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: deployment "istio-citadel" successfully rolled out
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: + kubectl rollout status deployment.extensions/istio-policy -n istio-system --timeout=120s
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: deployment "istio-policy" successfully rolled out
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: + kubectl rollout status deployment.extensions/istio-galley -n istio-system --timeout=120s
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: deployment "istio-galley" successfully rolled out
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: + kubectl rollout status deployment.extensions/istio-ingressgateway -n istio-system --timeout=120s
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: deployment "istio-ingressgateway" successfully rolled out
Dec 26 18:00:12 istio-my-cluster-vm startup-script: INFO startup-script: ++ echo -n admin
```

Reply:
+1